charms/worker/k8s/src/charm.py

```diff
@@ -149,6 +149,26 @@ def __init__(self, *args):
             self.etcd = EtcdReactiveRequires(self)
             self.framework.observe(self.on.get_kubeconfig_action, self._get_external_kubeconfig)
 
+        # kubernetes-info
+        if self.lead_control_plane:
+            self.framework.observe(self.on.ceph_k8s_info_relation_changed, self._k8s_info)
+
+    def _k8s_info(self, event: ops.RelationChangedEvent):
+        """Send cluster information on the kubernetes-info relation.
+
+        Provide applications with cluster characteristics. This should only run on the lead
+        k8s control plane unit.
+
+        Args:
+            event: ops.RelationChangedEvent - event triggered by the relation changed hook
+        """
+        if not self.lead_control_plane:
```

Comment:
This feels like a unnecessary double-check. 

I'd say either remove the outer or inner call since we do control the calling of this private function.

Reply:
agree... likely unnecessary depending on the caller.  If it's called from the reconcile function this isn't necessary